README.md

```diff
@@ -1,2 +1,112 @@
+[![Version](https://img.shields.io/cocoapods/v/CacheAdvance.svg)](https://cocoapods.org/pods/CacheAdvance)
+[![License](https://img.shields.io/cocoapods/l/CacheAdvance.svg)](https://cocoapods.org/pods/CacheAdvance)
+[![Platform](https://img.shields.io/cocoapods/p/CacheAdvance.svg)](https://cocoapods.org/pods/CacheAdvance)
+
 A cache that enables the performant persistence of individual messages to disk.
+
+## Usage
+
+### Basic Initialization
+
+```swift
+let myCache = try CacheAdvance<MyMessageType>(
+    file: FileManager.default.temporaryDirectory.appendingPathComponent("MyCache"),
+    maximumBytes: 5000,
+    shouldOverwriteOldMessages: false)
+```
+To begin caching messages, you need to create a CacheAdvance instance with:
+
+* A file URL – this URL must represent a file that has already been created. You can create a file by using `FileManager`'s [createFile(atPath:contents:attributes)](https://developer.apple.com/documentation/foundation/filemanager/1410695-createfile) API.
+* A maximum number of bytes on disk the cache can consume.
+* Whether the cache should overwrite old messages. If you need to preserve every message, set this value to `false`. If you care only about preserving recent messages, set this value to `true`.
+
+### Appending messages to disk
+
+```swift
+try myCache.append(message: aMessageInstance)
+```
+
+By the time the above method exits, the message will have been persisted to disk. A CacheAdvance keeps no in-memory buffer. Appending a new message is cheap, as a CacheAdvance needs to encode and persist only the new message and associated metadata.
+
+A CacheAdvance instance that does not overwrite old messages will throw a `CacheAdvanceError.messageDataTooLarge` if appending a message would exceed the cache's `maximumBytes`. A CacheAdvance instance that does overwrite old messages will throw a `CacheAdvanceError.messageDataTooLarge` if the message would require more than `maximumBytes` to store even after evicting all older messages from the cache.
+
+### Retrieving messages from disk
+
+```swift
+let cachedMessages = try myCache.messages()
+```
+
+This method reads all cached messages from disk into memory.
+
+### Thread safety
+
+CacheAdvances are not thread safe: a single CacheAdvance instance should always be interacted with from a single, serial queue. Since CacheAdvance reads from and writes to the disk synchronously, it is best to interact with a CacheAdvance on a background queue to prevent blocking the main queue.
+
+### Error handling
+
+A CacheAdvance will never fatal error: only recoverable errors will be thrown. A CacheAdvance may throw a `CacheAdvanceError`, or errors related to reading or writing with `FileHandle`s.
+
+If a `CacheAdvanceError.fileCorrupted` error is thrown, the cache file is corrupt and should be deleted.
+
+## How it works
+
+CacheAdvance immediately persists each appended messages to disk using `FileHandle`s. More information to come!
```

Comment:
Things we should talk about here:
* How messages are encoded
* File format

Reply:
https://github.com/dfed/CacheAdvance/pull/23/commits/6447d5590ae153effd92a9ee5235b1a487f04410